pylint/lint/pylinter.py

```diff
@@ -43,6 +43,14 @@ def _read_stdin():
     return sys.stdin.read()
 
 
+def _load_reporter_by_class(reporter_class: str) -> type:
+    qname = reporter_class
+    module_part = astroid.modutils.get_module_part(qname)
+    module = astroid.modutils.load_module_from_name(module_part)
+    class_name = qname.split(".")[-1]
+    return getattr(module, class_name)
```

Comment:
I checked the tests coverage and strangely it look like those three lines are not covered (?!). Do you have the same result on your side ?

Reply:
Ah, yes. This code loads custom reporter classes from their FQN. So instead of

```python
class MyReporter:
    name = "my_registered_name"
    ...

def register(linter):
    linter.register_reporter(MyReporter)

# pylint --load-plugins=my_plugin --output_format=my_registered_name
```

Users can do

```python
class MyReporter:
    ...

# pylint --output_format=my_plugin.MyReporter
```

Assuming that `my_plugin` is available somewhere on `PYTHONPATH`. This isn't my code. I just refactored it a bit. I can add a simple synthetic test case for this, but AFAIK, this didn't have any coverage before either.

Reply:
If you don't mind adding a test, that would be nice. To be fair I don't understand how the first two lines can be covered and not the last three.

Reply:
Will do.

I think, the first 2 lines are covered by `test_set_unsupported_reporter`. The third line raises an `ImportError` which gets promoted to an `InvalidReporterError`.